### ISSUE TITLE
Nextcloud inaccessible due to PHP Fatal error: Uncaught Error: Value of type null is not callable  from time to time

### ISSUE THREAD

**Issue by TheColin21** [nextcloud/server#27760]
### How to use GitHub

* Please use the 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to show that you are affected by the same issue.
* Please don't comment if you have no relevant information to add. It's just extra noise for everyone subscribed to this issue.
* Subscribe to receive notifications on status change and new comments. 

### Steps to reproduce
1. try to load Nextcloud
(Sorry, I know thats not very helpful but this error only appears sporadically.)

### Expected behaviour
Nextcloud opens

### Actual behaviour
An internal server error warning (by Nextcloud, not by the webserver) apperas.
### Server configuration

**Operating system:** Ubuntu 20.04

**Web server:** Apache2 behind Nginx reverse proxy

**Database:** MariaDB 10.5.11

**PHP version:** 8.0.7

**Nextcloud version:** 21.0.3

**Updated from an older Nextcloud/ownCloud or fresh install:** Updated

**Where did you install Nextcloud from:** nextcloud.com

**Signing status:**
<details>
<summary>Signing status</summary>
No errors have been found.
</details>

**List of activated apps:**
<details>
<summary>App list</summary>
Enabled:
  - accessibility: 1.7.0
  - activity: 2.14.3
  - admin_audit: 1.11.0
  - apporder: 0.13.0
  - breezedark: 21.0.8
  - camerarawpreviews: 0.7.11
  - checksum: 1.1.2
  - cloud_federation_api: 1.4.0
  - comments: 1.11.0
  - contactsinteraction: 1.2.0
  - dashboard: 7.1.0
  - dav: 1.17.1
  - documentserver_community: 0.1.9
  - extract: 1.3.1
  - federatedfilesharing: 1.11.0
  - federation: 1.11.0
  - files: 1.16.0
  - files_antivirus: 3.2.1
  - files_downloadactivity: 1.10.0
  - files_external: 1.12.0
  - files_fulltextsearch: 21.0.2
  - files_mindmap: 0.0.24
  - files_pdfviewer: 2.1.0
  - files_rightclick: 1.0.0
  - files_sharing: 1.13.1
  - files_texteditor: 2.14.0
  - files_trackdownloads: 1.10.0
  - files_trashbin: 1.11.0
  - files_versions: 1.14.0
  - files_videoplayer: 1.10.0
  - firstrunwizard: 2.10.0
  - fulltextsearch: 21.0.1
  - fulltextsearch_elasticsearch: 21.0.1
  - imageconverter: 1.3.1
  - impersonate: 1.8.0
  - logreader: 2.6.0
  - lookup_server_connector: 1.9.0
  - mail: 1.9.5
  - maps: 0.1.8
  - metadata: 0.13.0
  - notifications: 2.9.0
  - notify_push: 0.2.1
  - oauth2: 1.9.0
  - onlyoffice: 7.0.2
  - password_policy: 1.11.0
  - passwords: 2021.6.20
  - photos: 1.3.0
  - previewgenerator: 3.1.1
  - privacy: 1.5.0
  - provisioning_api: 1.11.0
  - ransomware_protection: 1.10.1
  - recommendations: 1.0.0
  - serverinfo: 1.11.0
  - settings: 1.3.0
  - sharebymail: 1.11.0
  - spreed: 11.2.2
  - support: 1.4.0
  - survey_client: 1.9.0
  - systemtags: 1.11.0
  - text: 3.2.0
  - theming: 1.12.0
  - twofactor_backupcodes: 1.10.0
  - updatenotification: 1.11.0
  - user_ldap: 1.11.0
  - user_status: 1.1.1
  - viewer: 1.5.0
  - weather_status: 1.1.0
  - workflowengine: 2.3.0
Disabled:
  - encryption
  - nextcloud_announcements
</details>

**Nextcloud configuration:**
<details>
<summary>Config report</summary>

{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "cloud.clnp.eu",
            "cloud.saef-zf.de"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "21.0.3.1",
        "overwrite.cli.url": "https:\/\/cloud.clnp.eu",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "mail_smtpmode": "sendmail",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "memcache.local": "\\OC\\Memcache\\APCu",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": 0
        },
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "updater.release.channel": "stable",
        "loglevel": 0,
        "maintenance": false,
        "theme": "",
        "app_install_overwrite": [
            "camerarawpreviews",
            "files_texteditor",
            "extract",
            "documentserver_community"
        ],
        "ldapIgnoreNamingRules": false,
        "ldapProviderFactory": "OCA\\User_LDAP\\LDAPProviderFactory",
        "default_language": "de",
        "default_locale": "de",
        "default_phone_region": "DE",
        "defaultapp": "files",
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***"
    }
}
</details>

**Are you using external storage, if yes which one:** Local, Nextcloud

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** no

### Client configuration
**Browser:** Firefox/Chrome/Edge, does not matter

**Operating system:** Windows 10, still does not matter

#### Web server error log
The following error, again and again:
<details>
<summary>Web server error log</summary>
AH01071: Got error 'PHP message: PHP Fatal error: Uncaught Error: Value of type null is not callable in /var/www/vhosts/domain.tld/cloud.domain.tld/3rdparty/phpseclib/phpseclib/phpseclib/Crypt/Base.php:868\nStack trace:\n#0 /var/www/vhosts/domain.tld/cloud.domain.tld/lib/private/Security/Crypto.php(100): phpseclib\\Crypt\\Base->encrypt()\n#1 /var/www/vhosts/domain.tld/cloud.domain.tld/lib/private/Session/CryptoSessionData.php(180): OC\\Security\\Crypto->encrypt()\n#2 /var/www/vhosts/domain.tld/cloud.domain.tld/lib/private/Session/CryptoSessionData.php(74): OC\\Session\\CryptoSessionData->close()\n#3 [internal function]: OC\\Session\\CryptoSessionData->__destruct()\n#4 {main}\n thrown in /var/www/vhosts/domain.tld/cloud.domain.tld/3rdparty/phpseclib/phpseclib/phpseclib/Crypt/Base.php on line 868'
</details>

#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>
nothing there regarding this issue as nextcloud cannot run while it lasts
</details>

Pretty much all there paths are inside the nextcloud folder. As you can see, this folder is part of the allowed paths

**Comment by sriptorium** [comment]
Exact same issue here. Not much to say. Can't figure it out...
Any luck fixing your instance in your case ?

**Comment by TheColin21** [comment]
No luck, sorry

Am 3. Juli 2021 23:07:59 MESZ schrieb efftee ***@***.***>:

**Comment by sriptorium** [comment]
Hi, 
I must add that since I had no fix I proceded to rollback to NC20+PHP7.4 (through a backup restore)
This is when I discovered that my SSD hard drive was failing.
I am unable to be certain if the "pre-fail" of that system SSD could have been at the origin of this error, but I just thought I'd make a mention about it here.

I will wait a few hours/days, before attempting a new upgrade to PHP8 and NC21 and keep you posted if the same issue reoccurs.
